edge-modules/TwinTester/Program.cs

```
@@ -0,0 +1,344 @@
+                    if (storageCount == 0)
+                    {
+                        Logger.LogInformation("No existing storage detected. Initializing new module twin for fresh run.");
+
+                        // reset desired properties
+                        Twin originalTwin = await registryManager.GetTwinAsync(Settings.Current.DeviceId, Settings.Current.ModuleId);
+                        Twin desiredPropertyResetTwin = await registryManager.ReplaceTwinAsync(Settings.Current.DeviceId, Settings.Current.ModuleId, new Twin(), originalTwin.ETag);
+
+                        // reset reported properties
+                        TwinCollection eraseReportedProperties = GetReportedPropertiesResetTwin(moduleClient, desiredPropertyResetTwin);
+                        await moduleClient.UpdateReportedPropertiesAsync(eraseReportedProperties);
+
+                        await Task.Delay(waitPeriodInMs);
+                    }
+
+                    Twin initializedTwin = await registryManager.GetTwinAsync(Settings.Current.DeviceId, Settings.Current.ModuleId);
+
+                    Logger.LogInformation($"Start state of module twin: {JsonConvert.SerializeObject(initializedTwin, Formatting.Indented)}");
+
+                    return initializedTwin.ETag;
+                }
+                catch (Exception e)
+                {
+                    Logger.LogInformation($"Retrying failed twin initialization: {e}");
+                    await Task.Delay(waitPeriodInMs);
+                }
+            }
+        }
+
+        static TwinCollection GetReportedPropertiesResetTwin(ModuleClient moduleClient, Twin originalTwin)
+        {
+            TwinCollection eraseReportedProperties = new TwinCollection();
+            foreach (dynamic twinUpdate in originalTwin.Properties.Reported)
```

I tried casting in the loop, but silently continued without iterating.